contracts/ModuleRegistry.sol

```diff
@@ -323,16 +350,42 @@ contract ModuleRegistry is IModuleRegistry, EternalStorage {
      * @return bool indicating whether module factory is verified
      * @return address array which contains the list of securityTokens that use that module factory
      */
-    function getFactoryDetails(address _factoryAddress) external view returns(bool, address[] memory) {
-        return (getBoolValue(Encoder.getKey("verified", _factoryAddress)), getArrayAddress(Encoder.getKey("reputation", _factoryAddress)));
+    function getFactoryDetails(address _factoryAddress) external view returns(bool, address, address[] memory) {
+        return (getBoolValue(Encoder.getKey("verified", _factoryAddress)), getAddressValue(Encoder.getKey("factoryOwner", _factoryAddress)), getArrayAddress(Encoder.getKey("reputation", _factoryAddress)));
     }
 
     /**
-     * @notice Returns the list of addresses of Module Factory of a particular type
+     * @notice Returns the list of addresses of verified Module Factory of a particular type
      * @param _moduleType Type of Module
      * @return address array that contains the list of addresses of module factory contracts.
      */
     function getModulesByType(uint8 _moduleType) public view returns(address[] memory) {
+        address[] memory _addressList = getArrayAddress(Encoder.getKey("moduleList", uint256(_moduleType)));
```

Comment:
This will only return modules with primary type (type[0]) == `_moduleType`. Maybe we should add the factory to all the relevant module type lists in registerModule()?

Reply:
Yeah - that would probably be better, although we'd also have to modify the `removeModule` logic and the "moduleListIndex" etc. to reflect this, so I'd say this is better in 3.1 than 3.0

Reply:
Agreed. As this function is really only for offchain stuff, I don't mind it returning incomplete data. Perhaps we should add a code comment about this though. I don't want dApp and CLI teams scratching their heads a few months down the line :).